### PR TITLE
add url to the notifications API

### PR DESCRIPTION
**Pull request by PetrDlouhy**
This MR adds URL of the target, action and actor url to the API endpoint (if the url is defined by `get_url_for_notifications(notification, request)` or `get_absolute_url()` method on the object).

We would like to make notifications loaded through the JS code with links to the target objects.

We wanted to add plain `get_absolute_url()` to the API endpoint, but we realized, that it is not sufficient in case where the link should depend also on the user that is accessing the notification, so we needed to add the `request` parameter (and we also added the notification object).

### CODE REVIEW

**Comment by PetrDlouhy** [comment]
@AlvaroLQueiroz Did you look at this PR? Should I add something to get this merged?

**Comment by PetrDlouhy** [comment]
@schinckel @yangyubo @AlvaroLQueiroz I have rebased this to the current master. Could you please take a look?

**Comment by schinckel** [comment]
That looks reasonable. I can't merge (or approve), but I'd suggest some tests would be useful.

**Review by nemesifier — COMMENTED** [review]
Definitely on a project like this one, used by many people, I am of the opinion that we need to ensure there's tests covering new cases like this one.

**Comment by PetrDlouhy** [comment]
@schinckel @nemesifier Aggree. I have added test for this and also a little documentation note.

File: notifications/tests/tests.py [hunk 1]
```diff
@@ -457,6 +458,31 @@ def test_unread_list_api_mark_as_read(self):  # pylint: disable=invalid-name
...
+        """
+        self.login('to', 'pwd')
+        Notification.objects.filter(recipient=self.to_user).mark_all_as_read()
+        to_customer = Customer.objects.create(name="to_customer")
+        action_customer = Customer.objects.create(name="action_customer")
+        from_customer = Customer.objects.create(name="from_customer")
+        target_object = TargetObject.objects.create(name="target_object")
+        notify.send(from_customer, recipient=self.to_user, verb='commented', action_object=action_customer,
+                    target=target_object)
+        response = self.client.get(reverse('notifications:live_unread_notification_list'))
```
Comment (nemesifier) [inline notifications/tests/tests.py:474]: I recommend maintaining consistency in the quoting style, some strings are quoted with single quotes, while other are quoted with double quotes. In most repositories I contribute/maintain we have the black formatter handling these format issues but here I guess we have to resort for manual checks unless we introduce black formatting enforcement in another PR.

File: notifications/tests/tests.py [hunk 2]
```diff
@@ -457,6 +458,31 @@ def test_unread_list_api_mark_as_read(self):  # pylint: disable=invalid-name
...
+        self.assertEqual(data['unread_count'], 1)
+        notification = data['unread_list'][0]
+        self.assertEqual(notification['actor'], "from_customer")
+        self.assertEqual(notification['action_object_url'], f'foo/{action_customer.id}/')
+        self.assertEqual(notification['target_url'], f'bar/{target_object.id}/')
+        self.assertEqual(notification['actor_url'], f'foo/{from_customer.id}/')
+
+
+
+
```
Comment (nemesifier) [inline notifications/tests/tests.py:485]: I would remove these blank lines and leave only 1 blank line

**Review by nemesifier — COMMENTED** [review]
Looks good, I left a couple of comments below for minor formatting improvements.

File: notifications/tests/tests.py (hunk 1 above)
Comment (PetrDlouhy) [reply notifications/tests/tests.py:474]: fixed

File: notifications/tests/tests.py (hunk 2 above)
Comment (PetrDlouhy) [reply notifications/tests/tests.py]: fixed

**Comment by PetrDlouhy** [comment]
@nemesifier Thanks. I have fixed it.

**Review by nemesifier — APPROVED** [review]
LGTM :heavy_check_mark:

**Comment by AlvaroLQueiroz** [comment]
I'm sorry for not getting back to you sooner. I will release the new version.

**Pull request merged**